elastica/timestepper/__init__.py

```diff
@@ -86,17 +81,15 @@ def integrate(
     assert final_time > 0.0, "Final time is negative!"
     assert n_steps > 0, "Number of integration steps is negative!"
 
-    # Extend the stepper's interface after introspecting the properties
-    # of the system. If system is a collection of small systems (whose
-    # states cannot be aggregated), then stepper now loops over the system
-    # state
-    do_step, stages_and_updates = extend_stepper_interface(StatefulStepper, System)
-
-    dt = np.float64(float(final_time) / n_steps)
-    time = restart_time
+    dt = np.float_(float(final_time) / n_steps)
+    time = np.float_(restart_time)
 
-    for i in tqdm(range(n_steps), disable=(not progress_bar)):
-        time = do_step(StatefulStepper, stages_and_updates, System, time, dt)
+    if is_system_a_collection(systems):
+        for i in tqdm(range(n_steps), disable=(not progress_bar)):
+            time = stepper.step(systems, time, dt)  # type: ignore[arg-type]
+    else:
+        for i in tqdm(range(n_steps), disable=(not progress_bar)):
+            time = stepper.step_single_instance(systems, time, dt)  # type: ignore[arg-type]
```

Comment:
Why do we need this step_single_instance call. Is this for tests?

Reply:
Seems like this is needed for Explicit steppers can you confirm.?

Reply:
`step_single_instance` is mostly used for testing at this point.